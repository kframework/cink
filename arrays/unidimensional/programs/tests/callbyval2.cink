void f(int a[], int i , int x) {
  int *p = &a[i];
  *p = x;
}

void main() {
  int b[7]; 
  f(b, 3, 5);
  cout << b[3];
}
