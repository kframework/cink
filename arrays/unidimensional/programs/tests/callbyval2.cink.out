5<T>
    <in>
        ListItem(#buffer ( "" ))
        ListItem(#istream ( 0 ))
    </in>
    <out>
        ListItem(#ostream ( 1 ))
        ListItem(#buffer ( "" ))
    </out>
    <control>
        <k>
            .K
        </k>
        <return>
            noVal
        </return>
        <fstack>
            .List
        </fstack>
        <sideEffects>
            .Bag
        </sideEffects>
        <env>
            f |-> #symScalLoc(1)
            main |-> #symScalLoc(2)
        </env>
    </control>
    <genv>
        f |-> #symScalLoc(1)
        main |-> #symScalLoc(2)
    </genv>
    <store>
        seq-array ( 7 , #symLoc(3) ) |-> store ( const-array ( 7 , 0 ) , 3
           , 5 )
        #symScalLoc(1) |-> lambda ( int (a []), int i, int x @ p of pointer to
           int ; (p = (& (a [ i ]))) ; ((* p) = x) ; )
        #symScalLoc(2) |-> lambda (  @ int (b [ 7 ]) ; (f ( (b , (3 , 5)) )) ;
           (cout << (b [ 3 ])) ; )
        #symScalLoc(4) |-> 3
        #symScalLoc(5) |-> 5
        #symScalLoc(6) |-> select ( seq-array ( 7 , #symLoc(3) ) , 3 )
    </store>
</T>
